cpp/proto/serializer.h

```diff
@@ -10,34 +10,57 @@
 #include "base/macros.h"
 #include "proto/ct.pb.h"
 
+
+// Serialization methods return OK on success,
+// or the first encountered error on failure.
+enum class SerializeResult {
+  OK,
+  INVALID_ENTRY_TYPE,
+  EMPTY_CERTIFICATE,
+  // TODO(alcutter): rename these to LEAFDATA_TOO_LONG or similar?
+  CERTIFICATE_TOO_LONG,
+  CERTIFICATE_CHAIN_TOO_LONG,
+  INVALID_HASH_ALGORITHM,
+  INVALID_SIGNATURE_ALGORITHM,
+  SIGNATURE_TOO_LONG,
+  INVALID_HASH_LENGTH,
+  EMPTY_PRECERTIFICATE_CHAIN,
+  UNSUPPORTED_VERSION,
+  EXTENSIONS_TOO_LONG,
+  INVALID_KEYID_LENGTH,
+  EMPTY_LIST,
+  EMPTY_ELEM_IN_LIST,
+  LIST_ELEM_TOO_LONG,
+  LIST_TOO_LONG,
+  EXTENSIONS_NOT_ORDERED,
+};
+
+std::ostream& operator<<(std::ostream& stream, const SerializeResult& r);
```

Comment:
They're present, but github seems not to want to show the diffs in `serializer.cc`, it says:
"341 additions, 280 deletions not shown" for that file, not entirely sure why...